src/gossip/service.zig

```diff
@@ -417,26 +410,23 @@ pub const GossipService = struct {
 
             // verify in parallel using the threadpool
             // PERF: investigate CPU pinning
-            var task_i: usize = 0;
-            const n_tasks = tasks.len;
+            var task_search_start_idx: usize = 0;
             for (packet_batches) |packet_batch| {
-                // find a free task
-                var task_ptr = &tasks[task_i];
-                while (!task_ptr.done.load(.acquire)) {
-                    task_i = (task_i + 1) % n_tasks;
-                    task_ptr = &tasks[task_i];
-                }
-                // schedule it
-                task_ptr.done.store(false, .release);
-                task_ptr.packet_batch = packet_batch;
+                const acquired_task_idx = VerifyMessageTask.awaitAndAcquireFirstAvailableTask(tasks, task_search_start_idx);
+                task_search_start_idx = (acquired_task_idx + 1) % tasks.len;
+
+                const task_ptr = &tasks[acquired_task_idx];
+                task_ptr.entry.packet_batch = packet_batch;
+                task_ptr.result catch |err| self.logger.errf("VerifyMessageTask encountered error: {s}", .{@errorName(err)});
 
                 const batch = Batch.from(&task_ptr.task);
                 self.thread_pool.schedule(batch);
```

Comment:
i feel like these two are always used together - can we throw them in a helper fcn of `ThreadPoolTask`

Reply:
I would prefer to keep the `thread_pool.schedule` invocation top-level, not too hidden down in helper functions. I think it would be reasonable to have a method on ThreadPoolTask to make a batch out of it though, making it `self.thread_pool.schedule(task_ptr.toBatch())`.